src/blockencodings.h

```diff
@@ -12,58 +12,70 @@
 class CTxMemPool;
 
 // Dumb helper to handle CTransaction compression at serialize-time
-struct TransactionCompressor {
+template<typename T>
+struct TransactionCompressWrapper {
 private:
-    CTransactionRef& tx;
+    T& tx;
 public:
-    explicit TransactionCompressor(CTransactionRef& txIn) : tx(txIn) {}
-
-    ADD_SERIALIZE_METHODS;
+    explicit TransactionCompressWrapper(T& txIn) : tx(txIn) {}
+    SERIALIZE_METHODS(obj) { READWRITE(obj.tx); }
+};
 
-    template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream& s, Operation ser_action) {
-        READWRITE(tx); //TODO: Compress tx encoding
+template<typename I>
+struct Uint48Wrapper {
+private:
+    I& m_int;
+public:
+    Uint48Wrapper(I& i) : m_int(i) {}
+    template <typename Stream> void Serialize(Stream& s) const
+    {
+        uint32_t lsb = m_int & 0xffffffff;
+        uint16_t msb = (m_int >> 32) & 0xffff;
+        s << lsb << msb;
+    }
+    template <typename Stream> void Unserialize(Stream& s)
+    {
+        uint32_t lsb;
+        uint16_t msb;
+        s >> lsb >> msb;
+        m_int = (uint64_t(msb) << 32) | uint64_t(lsb);
     }
 };
 
+template<typename T>
```

Comment:
In commit "Convert blockencodings to new serialization"

Maybe move this up closer to TransactionCompressWrapper class

Reply:
Done.